metaserver/src/main/scala/scala/meta/languageserver/Semanticdbs.scala

```diff
@@ -4,8 +4,25 @@ import java.nio.file.Files
 import com.typesafe.scalalogging.LazyLogging
 import org.langmeta.internal.semanticdb.schema.Database
 import org.langmeta.io.AbsolutePath
+import scala.meta.interactive.InteractiveSemanticdb
+import scala.meta.parsers.ParseException
+import scala.tools.nsc.interactive.Global
+import langserver.types.VersionedTextDocumentIdentifier
+import ScalametaEnrichments._
 
 object Semanticdbs extends LazyLogging {
+  def loadFromTextDocument(
+    compiler: Global,
+    td: VersionedTextDocumentIdentifier,
+    content: String
+  ): scala.meta.semanticdb.Database = {
```

Comment:
Not sure about the difference between `schema.Database` and `semanticdb.Database`, but since `InteractiveSemanticdb.toDocument` returns a `semanticdb.Document` it seemed natural to return directly a `semanticdb.Database`.

@olafurpg Is there a reason why `loadFromFile` returns a `schema.Database` instead?

Reply:
semanticdb.Database is really slow to load while schema.Database is much faster. We are slowly moving to schema.Database, scalafix currently depends on semanticdb.Database but we can make it do the conversion from schema.Database to semanticdb.Database lazily instead.

Reply:
Is there a way to build a `schema.Database` given a `semanticdb.Document`?

Reply:
There is a `.toSchema` extension method on `semanticdb.Database`, you may need an import but I'm not sure, see https://github.com/scalameta/scalameta/blob/dc33d1f66bea57424d0734628c38d12d76344887/langmeta/shared/src/main/scala/org/langmeta/semanticdb/Database.scala#L18